mon-pix/app/components/signup-form.js

```diff
@@ -188,4 +208,25 @@ export default class SignupForm extends Component {
     };
     return this.intl.t(httpStatusCodeMessages[statusCode] || httpStatusCodeMessages['default']);
   }
+
+  _getLocaleFromCookieOrDomain() {
+    const currentDomainExtension = this.currentDomain.getExtension();
+    const localeCookie = this.cookies.read('locale');
+    const currentLocale = this.intl.get('locale')[0];
+    const canonicalLocale = Intl.getCanonicalLocales(currentLocale)[0];
+
+    if (currentDomainExtension === INTERNATIONAL_DOMAIN_TLD) {
+      if (localeCookie) {
+        return localeCookie;
+      }
+
+      return canonicalLocale;
+    }
+
+    if (currentDomainExtension === FRENCH_DOMAIN_TLD) {
+      return FRENCH_LOCALE;
+    }
+
+    return canonicalLocale;
+  }
```

Comment:
Suggestion de code avec les modifications suivantes : 
* renommage `localeCookie`→`cookieLocale` car il s'agit d'une locale et non d'un cookie
* suppression de la canonicalisation de la valeur de la locale car le code le fait déjà côté serveur

```suggestion
  _getLocaleFromCookieOrDomain() {
    const currentDomainExtension = this.currentDomain.getExtension();
    const cookieLocale = this.cookies.read('locale');
    const currentLocale = this.intl.get('locale')[0];

    if (currentDomainExtension === INTERNATIONAL_DOMAIN_TLD) {
      if (cookieLocale) {
        return cookieLocale;
      }

      return currentLocale;
    }

    if (currentDomainExtension === FRENCH_DOMAIN_TLD) {
      return FRENCH_LOCALE;
    }

    return currentLocale;
  }
```